libexec/revng/revng-ptml

```diff
@@ -0,0 +1,267 @@
+
+    filters = normalize_filter_extract(args.filter, args.extract)
+    content = args.input.read()
+    result = handle_file(content, strip_ptml, strip_ptml_many, filters)
+    if args.inplace:
+        args.input.seek(0)
+        args.input.truncate(0)
+        args.input.write(result)
+    else:
+        args.output.write(result)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Tool to manipulate PTML files")
+    subparsers = parser.add_subparsers(title="Subcommands", metavar="")
+
+    parser_cat = subparsers.add_parser("cat", help="Print the ptml on the console")
+    parser_cat_filter = parser_cat.add_mutually_exclusive_group()
+    parser_cat_filter.add_argument(
+        "-f",
+        "--filter",
+        type=str,
+        action="append",
+        default=[],
+        required=False,
+        help="Only show the keys specified (if present)",
+    )
+    parser_cat_filter.add_argument(
+        "-e", "--extract", type=str, required=False, help="Extract the specified key"
+    )
+    parser_cat_color = parser_cat.add_mutually_exclusive_group()
+    parser_cat_color.add_argument(
+        "-p", "--plain", action="store_true", help="Force plaintext output"
+    )
+    parser_cat_color.add_argument("-c", "--color", action="store_true", help="Force color output")
+    parser_cat.add_argument(
+        "input",
+        type=argparse.FileType("r"),
+        default=sys.stdin,
+        nargs="?",
+        help="Input file (stdin if omitted)",
+    )
+    parser_cat.set_defaults(func=cmd_cat)
+
+    parser_strip = subparsers.add_parser("strip", help="Strip PTML from a file")
+    parser_strip_filter = parser_strip.add_mutually_exclusive_group()
+    parser_strip_filter.add_argument(
+        "-f",
+        "--filter",
+        type=str,
+        action="append",
+        default=[],
+        required=False,
+        help="Only show the keys specified (if present)",
+    )
+    parser_strip_filter.add_argument(
+        "-e", "--extract", type=str, required=False, help="Extract the specified key"
+    )
+    parser_strip.add_argument(
+        "input",
+        type=argparse.FileType("r+"),
+        default=sys.stdin,
+        nargs="?",
+        help="Input file (stdin if omitted)",
+    )
+    parser_strip_out = parser_strip.add_mutually_exclusive_group()
+    parser_strip_out.add_argument("-i", "--inplace", action="store_true", help="Strip inplace")
+    parser_strip_out.add_argument(
+        "output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        nargs="?",
+        help="Output file (stdout if omitted)",
+    )
+    parser_strip.set_defaults(func=cmd_strip)
+
+    args = parser.parse_args()
+    args.func(args)
+
+
+if __name__ == "__main__":
+
+    def new_unraisablehook(arg):
+        if arg.exc_type != BrokenPipeError:
+            sys.__unraisablehook__(arg)
+
+    sys.unraisablehook = new_unraisablehook
+
+    with suppress(BrokenPipeError, KeyboardInterrupt):
+        main()
```

Comment:
Reasons to make this a standalone script as oppose to be a `revng.cli.commands_registry.Command`?

Reply:
I'd argue that since PTML will be "shipped" to the clients, it might be nice to have a standalone tool rather then have to drag the entirety of revng with it

Reply:
Uhm, this calls for a more general question: what do we put in the `pip install`-able package?
In theory we could put a subset of those made available in `revng-distributable`.

Reply:
Still, even in the case we have it in a pip package it would be a hassle to section off part of the `revng` command rather than `cp` the script

Reply:
I think this problem will pop up with more tools and shipping them as standalone tools won't be a good long term solution.
I think in `revng.py` we should detect if `libexec/revng` exists and if not disable a bunch of commands.

Reply:
Fixed